documentation/config_file_auth_migration.markdown

```diff
@@ -0,0 +1,342 @@
+            name:
+        },
+    ...
+    ```
+
+2. Next, add its type to the section's [`type`](./config_file_auth.markdown#match-request) setting. Since this is a literal string path, the type is `path`.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+3. The legacy rule has a [`method`](./config_file_auth.markdown#method-1) setting, with an indirector value of `find` that's equivalent to the GET and POST HTTP methods. We can implement these by adding an optional HOCON [`method`](./config_file_auth.markdown#match-request) setting in the rule's `match-request` section and specifying GET and POST as an array.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+4. Next, set the [`allow`](#allow-allow-unauthenticated-and-deny) setting. The legacy rule used a `*` glob, which is also supported in HOCON.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: "*"
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+5. Finally, give the rule a unique [`name`](#name) value. Remember that the rule will appear in logs and in the body of error responses to unauthorized clients.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: "*"
+            sort-order: 1
+            name: "environments"
+        },
+    ...
+    ```
+
+Our HOCON `auth.conf` file should now allow all authenticated clients to make GET and POST requests to the `/puppet/v3/environments` endpoint, and should look like this:
+
+``` hocon
+authorization: {
+    version: 1
+    rules: [
+        {
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: "*"
+            sort-order: 1
+            name: "environments"
+        },
+    ]
+}
+```
+
+### Converting more complex rules
+
+#### Paths set by regular expressions
+
+To convert a regular expression path, enclose it in double quotation marks and solidus characters (`/`), and set the `type` to regex.
+
+> **Note:** You must escape regular expressions to conform to HOCON standards, which are the same as JSON's and differ from the deprecated Ruby format's regular expressions. For instance, the digit-matching regular expression `\d` must be escaped with a second backslash, as `\\d`.
```

Comment:
A minor nitpick - I think it's probably preferable to call the old format "legacy" and not "Ruby".  We don't want to hate on Ruby too much and in reality users shouldn't really need to know implementation language, just old vs. new.